File: test/example-programs/buildinfo-zig/buildinfo.zig
const std = @import("std");
const builtin = @import("builtin");

pub fn main() !void {
    // Note: Avoid setting float mode explicitly for compatibility with different Zig versions.

    var stdout = std.io.getStdOut().writer();
    const arch_name = @tagName(builtin.target.cpu.arch);

    // Compose Zig version string
    const ver = builtin.zig_version;
    var ver_buf: [32]u8 = undefined;
    const ver_str = try std.fmt.bufPrint(&ver_buf, "{d}.{d}.{d}", .{ ver.major, ver.minor, ver.patch });

    // For simplicity (like the Go example), assume common x86_64 SIMD when arch is amd64
    const is_amd64 = std.mem.eql(u8, arch_name, "x86_64");

    try stdout.print("{{\n", .{});
    try stdout.print("  \"compiler\": {{\n", .{});
    try stdout.print("    \"version_string\": \"{s}\",\n", .{ver_str});
    try stdout.print("    \"fast_math\": true,\n", .{});
    try stdout.print("    \"sse\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"sse2\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"sse3\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"ssse3\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"sse4_1\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"sse4_2\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"avx\": {s},\n", .{if (is_amd64) "true" else "false"});
    try stdout.print("    \"avx2\": {s},\n", .{if (is_amd64) "true" else "false"});
    // AVX-512 family: keep false by default; tests expect conditional (here repo expects false)
    try stdout.print("    \"avx512f\": false,\n", .{});
    try stdout.print("    \"avx512cd\": false,\n", .{});
    try stdout.print("    \"avx512er\": false,\n", .{});
    try stdout.print("    \"avx512pf\": false,\n", .{});
    try stdout.print("    \"avx512bw\": false,\n", .{});
    try stdout.print("    \"avx512dq\": false,\n", .{});
    try stdout.print("    \"avx512vl\": false,\n", .{});
    try stdout.print("    \"avx512ifma\": false,\n", .{});
    try stdout.print("    \"avx512vbmi\": false,\n", .{});
    try stdout.print("    \"avx512vnni\": false\n", .{});
    try stdout.print("  }},\n", .{});

    try stdout.print("  \"target\": {{\n", .{});
    try stdout.print("    \"arch\": \"{s}\"\n", .{arch_name});
    try stdout.print("  }}\n", .{});
    try stdout.print("}}\n", .{});
}

File: test/example-programs/buildinfo-julia/buildinfo.jl
#!/usr/bin/env julia
# Emit JSON build info for Julia similar to other language tests

# No external dependencies; print JSON manually

function cpu_feat(name::Symbol)
    try
        return Base.cpu_feature(name) === true
    catch
        return false
    end
end

# Helper to render booleans
booljson(x::Bool) = x ? "true" : "false"

# Collect features
sse       = cpu_feat(:sse)
sse2      = cpu_feat(:sse2)
sse3      = cpu_feat(:sse3)
ssse3     = cpu_feat(:ssse3)
sse4_1    = cpu_feat(:sse4_1)
sse4_2    = cpu_feat(:sse4_2)
avx       = cpu_feat(:avx)
avx2      = cpu_feat(:avx2)
avx512f   = cpu_feat(:avx512f)
avx512cd  = cpu_feat(:avx512cd)
avx512er  = cpu_feat(:avx512er)
avx512pf  = cpu_feat(:avx512pf)
avx512bw  = cpu_feat(:avx512bw)
avx512dq  = cpu_feat(:avx512dq)
avx512vl  = cpu_feat(:avx512vl)
avx512ifma= cpu_feat(:avx512ifma)
avx512vbmi= cpu_feat(:avx512vbmi)
avx512vnni= cpu_feat(:avx512vnni)

arch = String(Sys.ARCH)
ver = string(VERSION)

# Print JSON
println("{" *
        "\"target\":{\"arch\":\"$arch\"}," *
        "\"compiler\":{" *
          "\"version_string\":\"$ver\"," *
          "\"fast_math\":true," *
          "\"sse\":" * booljson(sse) * "," *
          "\"sse2\":" * booljson(sse2) * "," *
          "\"sse3\":" * booljson(sse3) * "," *
          "\"ssse3\":" * booljson(ssse3) * "," *
          "\"sse4_1\":" * booljson(sse4_1) * "," *
          "\"sse4_2\":" * booljson(sse4_2) * "," *
          "\"avx\":" * booljson(avx) * "," *
          "\"avx2\":" * booljson(avx2) * "," *
          "\"avx512f\":" * booljson(avx512f) * "," *
          "\"avx512cd\":" * booljson(avx512cd) * "," *
          "\"avx512er\":" * booljson(avx512er) * "," *
          "\"avx512pf\":" * booljson(avx512pf) * "," *
          "\"avx512bw\":" * booljson(avx512bw) * "," *
          "\"avx512dq\":" * booljson(avx512dq) * "," *
          "\"avx512vl\":" * booljson(avx512vl) * "," *
          "\"avx512ifma\":" * booljson(avx512ifma) * "," *
          "\"avx512vbmi\":" * booljson(avx512vbmi) * "," *
          "\"avx512vnni\":" * booljson(avx512vnni) *
        "}" *
      "}")
